### ISSUE TITLE
clustering algorithm not ideal

### ISSUE THREAD

**Issue by ISteampowered** [dodona-edu/dolos#39]
The current clustering algorithm, which is based on the union-find algorithm, groups matches together even when the matches do not make the criteria to belong to the cluster. For example we have four files A, B, C and D and the matches between these are: 

- A with B
- B with C
- C with D
- A with D

Let's say the first three have a high enough score to be grouped together, then we will have a single equivalence group. With all the four files in them. Now the fourth and final match does not have a high enough score for the two files to be considered equal, so nothing changes to the equivalent groups. 
This is all as expected within union find. However this forms a problem for us as the fourth match will be grouped together with the first three despite it not having a high enough score to do so.
This will cause problems when one student copies many code fragments from many different people.
TL;DR: algorithm assumes that there is an equivalence relationship when there isn't one.

**Comment by ISteampowered** [comment]
Defining the equivalence relationship on ranges instead of files would not fix this issue, and would make the results harder to read.

**Comment by ISteampowered** [comment]
An alternative is that a graph showing the relationships between the files to navigate to the comparison pages.

**Comment by ISteampowered** [comment]
@bmesuere

**Comment by rien** [comment]
Since we're currently not clustering anymore, I'm closing this issue.
